willow/plugins/opencv/base.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import absolute_import
+
+import os
+
+from willow.image import Image
+
+
+class BaseOpenCVImage(Image):
+    def __init__(self, image, size):
+        self.image = image
+        self.size = size
+
+    @classmethod
+    def check(cls):
+        raise ImportError()
+
+    @Image.operation
+    def get_size(self):
+        return self.size
+
+    @Image.operation
+    def has_alpha(self):
+        # Alpha is not supported by OpenCV
+        return False
+
+    @Image.operation
+    def has_animation(self):
+        # Animation is not supported by OpenCV
+        return False
+
+    def _find_cascade(self, cascade_filename):
+        # If a relative path was provided, check local cascades directory
+        if not os.path.isabs(cascade_filename):
+            cascade_filename = os.path.join(
+                    os.path.dirname(os.path.dirname(__file__)),
+                    '..',
+                    'data/cascades',
+                    cascade_filename,
+            )
+        return cascade_filename
+
+
+class BaseOpenCVColorImage(BaseOpenCVImage):
+    @classmethod
+    def from_buffer_rgb(cls, image_buffer):
+        raise NotImplemented()
+
+    # TODO: Converter back to RGBImageBuffer
+
+
+class BaseOpenCVGrayscaleImage(BaseOpenCVImage):
+    min_size = (40, 40)
```

Comment:
These attribute names don't make much sence outside of the ``detect_faces`` method. Could you move them back or change their names to make their use more clear? (eg ``min_face_size``)